core/message-tags-3.3.md

```diff
@@ -105,7 +105,9 @@ Servers MAY apply moderation to this command using existing or newly specified m
 
 Servers MUST NOT deliver `TAGMSG` to clients that haven't negotiated the message tags capability.
 
-Servers SHOULD reject any `TAGMSG` command sent without tags. In this case, they MUST use the `ERR_NEEDMOREPARAMS` (`461`) error numeric.
+Servers SHOULD reject any `TAGMSG` command sent without tags, or where there are no tags with the client-only prefix
+and all tags without the client-only prefix are removed by the server.
```

Comment:
Delivering empty TAGMSG isn't a problem for clients. The messages are inert without tags. That said I don't feel too strongly about whether this is SHOULD or MUST territory. I think I was erring on the side of allowing server implementations more leeway for something harmless, but I can't think of a good reason to allow it.